File: MinerViewer.Data/Entities/Miner.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MinerViewer.Data.Entities
{
    public class Miner
    {
        public int Id { get; set; }
        public string? Name { get; set; }
        public string? Os { get; set; }
        public string? Coin { get; set; }
        public string? Location { get; set; }
        public List<GraphicCard>? Cards { get; set; }
        public string? Ip { get; set; }
    }
}

File: MinerViewer/ClientApp/src/app/Data/miner.ts
export interface Miner {
    id: number;
    name?: string;
    os?: string;
    coin?: string;
    location?: string;
    cards: GraphicCard[];
    ip?: string;
}

export interface GraphicCard {
    id: number;
    name?:string;
    memoryTemp?:number;
    hotSpotTemp?:number;
    cooler?:string;
}

export interface MinerAdress {
    id:number;
    ip:string;
}